### DISCUSSION TITLE
Why do we need a second editor instance

### DISCUSSION THREAD

**Discussion by iamleson98** [umaranis/svelte-lexical#90] [Q&A] [answered]
Hello sir, @umaranis 

I'm having 2 questions, so please answer me if you can. I appreciate.

First I need to admit that your code looks clean and well written.
I have got some good ideas on lexical and svelte itself after reading your code.

question 1: Why do we need 2 instances of Lexical Editor class ? `editor` and `activeEditor`
question 2: Why do you put those editor instances in both svelte store and context ?
I'm thinking that we just need stores for them, and that's  enough

Thank you

**Comment by umaranis** [comment, marked as answer]
Hi @iamleson98 ,

Both are excellent questions.

1. Lexical supports nested editors. So you can have a child editor inside the main editor. For instance, the image label is a nested editor and it can have its own separate configuration. `editor` references the main editor while `activeEditor` references the active editor that could be the main editor or an editor nested inside it.
2. To support multiple editors on the same page, we need to put editor references into the context so that child components are able to pick up the correct parent.
